pytorch_lightning/trainer/training_loop.py

```diff
@@ -467,7 +467,7 @@ def run_training_epoch(self):
             # only track outputs when user implements training_epoch_end
             # otherwise we will build up unnecessary memory
             if self.is_overridden('training_epoch_end', model=self.get_model()):
-                epoch_output.append(batch_output.training_step_output_for_epoch_end)
+                epoch_output += batch_output.training_step_output_for_epoch_end
```

Comment:
this is equivalent... why do we need to change it?

Reply:
It will create a list of list dict outputs(i.e. [[{},{}], [{},{}]]). However, += will create only the list of dict outputs

Reply:
Will, it is not equal as for the original it adds element the new merge lists... Equivalent would be
```
a = [1]
b = 2
a.append(b)
b += [b]
```